### PR TITLE
Update typeahead  to v0.11.1

### LINKED ISSUE #282: Update typeahead to the latest version

### PR DESCRIPTION
**Pull request by fredj**
This change is not backward compatible

Fixes #282.

### CODE REVIEW

File: externs/typeahead.js [hunk 1]
```diff
@@ -151,22 +151,23 @@ var TypeaheadDataset;
...
 jQuery.prototype.typeahead = function(options, var_dataset) {};
 
 
 /**
- * @param {string|TypeaheadOptions} options
- * @param {...TypeaheadDataset} var_dataset
- * @return {!angular.JQLite}
+ * @param {TypeaheadOptions|string} options
+ * @param {...TypeaheadDataset|string} var_dataset
+ * @return {!jQuery}
```
Comment (elemoine) [inline externs/typeahead.js]: Are you sure about this change (from `angular.JQLite` to `jQuery`)? Note that there is `jQuery.prototype.typeahead` when jQuery is used.

Comment (fredj) [inline externs/typeahead.js]: you're right, it's an error

**Comment by fredj** [comment]
This change is not backward compatible because the css class names have changed

**Comment by ger-benjamin** [comment]
(It's the *default* new css classname because typeahead offer to you to rename them. But in every case, events names have changed, so that' not backward compatible)
RTM for me

**Comment by fredj** [comment]
Geoportail-Luxembourg will be affected but it's not a huge change: [search.less](https://github.com/Geoportail-Luxembourg/geoportailv3/blob/master/geoportailv3/static/less/search.less) needs to be updated

**Comment by elemoine** [comment]
Can we postpone the merging of this PR? I'd like to be able to update the Luxembourg project when this is merged.

**Comment by fredj** [comment]
@elemoine please merge this PR when you are ready

File: examples/search.html [hunk 2]
```diff
@@ -45,17 +45,13 @@
...
+      span.twitter-typeahead .tt-suggestion:focus {
         color: #ffffff;
         text-decoration: none;
         outline: 0;
         background-color: #428bca;
       }
-      span.twitter-typeahead .tt-suggestion.tt-cursor {
-        color: #ffffff;
-        background-color: #428bca;
-      }
```
Comment (elemoine) [inline examples/search.html]: Why is this removed completely?

Comment (fredj) [inline examples/search.html]: It's a mistake, I'll fix it

Comment (fredj) [inline examples/search.html]: fixed

**Comment by elemoine** [comment]
Thanks @fredj. I'll soon merge this.

**Comment by elemoine** [comment]
For posterity: when updating the Luxembourg project to use typeahead.js 0.11.1 I hit this bug in typeahead.js: https://github.com/twitter/typeahead.js/pull/1319. I hope this will be fixed in typeahead.js 0.11.2.
